server/src/main/java/org/elasticsearch/index/engine/EngineConfig.java

```diff
@@ -128,6 +128,8 @@ public Supplier<RetentionLeases> retentionLeasesSupplier() {
     @Nullable
     private final Engine.IndexCommitListener indexCommitListener;
 
+    private final boolean recoveringAsPrimary;
```

Comment:
Can we document this and the getter too, indicating that it captures the primary state when recovering starts , is not updated, should only be used for stateless as a temporary solution and should only be used during recovery?

Reply:
I've added a comment